libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2154,7 +2154,10 @@ void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCa
 		return;
 	}
 
-	solAssert(!functionPointerType->takesArbitraryParameters(), "Function must have fixed parameters.");
+	solAssert(functionPointerType->kind() != FunctionType::Kind::DelegateCall);
+	auto const externallyCallableFunction = functionPointerType->asExternallyCallableFunction(false);
```

Comment:
Just a though, might it be better to overwrite `functionPointerType` with the result of this call instead? It is not used afterwards but could lead to accidentally be used in the future if one is not careful.

Reply:
Not sure about others but personally I prefer choosing names that are distinct enough and keeping the original variable over overwriting the same variable with something slightly different. I actually wish variables were `const` by default, which would somewhat discourage overwriting them. You might still want to access the previous value later or you could move a chunk of code not realizing that at the new location the variable has a new meaning.

Reply:
If you need to access something later, you can always change it back, in that case it is at least a very conscious decision to do so.
An alternative could be to never declare `functionPointerType` in the function scope but a tiny sub-scope where the only thing it does is assigning to an `externallyCallableFunction` (or however you want to call it) variable in the function scope.
As before, if you DO need to access it at a later time, you can always move it out of the scope.

Reply:
> An alternative could be to never declare `functionPointerType` in the function scope

Yeah, I agree, this would be much better.